docs/reference/module-options.md

```diff
@@ -0,0 +1,139 @@
+# Module configuration options
+
+Apostrophe modules can be configured with settings that influence functionality without having to write custom Javascript. The sections below describe the options available in all modules as well as those specific to certain module types.
+
+All settings described here are placed in a module's `options` configuration object. The `options` object can be added in the module's `index.js` file, as well as where the module is instantiated in the `app.js` file.
+
+`index.js` example:
+```javascript
+// modules/article/index.js
+module.exports = {
+  extend: '@apostrophecms/piece-type',
+  // 👇 Module configuration options
+  options: {
+    alias: 'article',
+    label: 'Article',
+    pluralLabel: 'Articles',
+    quickCreate: true
+  },
+  // Other settings, such as `fields`
+}
+```
+
+`app.js` example:
+```javascript
+// app.js
+require('apostrophe')({
+  shortName: 'bowling-league-site', // Unique to your project
+  modules: {
+    article: {
+      extend: '@apostrophecms/piece-type',
+      // 👇 Module configuration options
+      options: {
+        alias: 'article',
+        label: 'Article',
+        pluralLabel: 'Articles',
+        quickCreate: true
+      },
+      // Other settings, such as `fields`
+    }
+  }
+});
+```
+
+## Options for any module
+
+- [`alias`](#alias)
+- [`components`](#components)
+- [`name`](#name)
+- [`templateData`](#templatedata)
+
+### `alias`
+
+<!-- TODO: Link to information about the apos object when available. -->
+Set to a string, the `alias` value will be applied to the `apos` object (accessible in many places) as a quick reference to the module.
+set an alias to easily reference the module from other modules. There is no default value.
+
+#### Example
+
+```javascript
+// modules/news-article/index.js
+module.exports = {
+  extend: '@apostrophecms/piece-type',
+  options: {
+    alias: 'article'
+  },
+  // ...
+}
+```
+
+This `article` module can then be referenced anywhere the `apos` object is present as `apos.article`. Otherwise it would be available via `apos.modules['news-article']`.
+
+
+### `components`
+
+The `components` options is an object identifying Vue components to use for the module's related user interface. The keys of this object, and thus the UI being overridden, will vary based on the module type. For example, piece module use `managerModal` and `editorModal` components.
+
+This is an advanced option since it can easily break the user interface.
+
+#### Example
+
+```javascript
+// modules/@apostrophecms/piece-type/index.js
+module.exports = {
+  options: {
+    components: {
+      managerModal: 'MyCustomPiecesManager'
+    }
+  },
+  // ...
+}
+```
+
+### `name`
+
+Modules' `name` properties are based on the `module` object keys in `app.js` by default. Setting the the `name` option in a module can override this original value for the sake of referencing the module elsewhere.
```

Comment:
The purpose of `name` has historically been to set the string used as a `type` property for docs of that type. In 3.x it is by default always the same as the module name, and I think we should stomp out `self.options.name` in favor of setting `self.name` firmly to `self.__meta.name` rather than just as a default. It can only cause confusion.

Reply:
that works for me.